nixos/modules/services/misc/sonarr.nix

```diff
@@ -48,26 +82,41 @@ in
       "d '${cfg.dataDir}' 0700 ${cfg.user} ${cfg.group} - -"
     ];
 
-    systemd.services.sonarr = {
-      description = "Sonarr";
-      after = [ "network.target" ];
-      wantedBy = [ "multi-user.target" ];
+    systemd.services.sonarr =
+      let
+        # add empty ApiKey so it can be replaced afterwards
+        configContent =
+          settingsCombined // (lib.optionalAttrs (cfg.apiKeyFile != "") { ApiKey = "#APIKEY#"; });
+        configFile = settingsFormat.generate "sonarr-config.xml" { Config = configContent; };
+      in
```

Comment:
Nit/personal optinion (you're free to ignore): I'd move the contents of this let/in to the top let/in. It creates a bigger scope, yes, but I find it easier to read, and has less churn/whitespace in the service definition.

Reply:
done